app/models/pawn.rb

```diff
@@ -6,4 +6,39 @@ def image
       '&#9823;'
     end
   end
+
+  def valid_move?(x, y)
+    dx = delta_x(x_position, x)
+    dy = delta_y(y_position, y)
+
+    return false if blocked?(x, y)
+
+    # pawn can't move horizontally
+    return false if dx != 0 && dy.zero?
+
+    # pawn moves vertically 1 space
+    return true if dx.zero? && dy == 1
+    return false if color == 'white' && y_position < y
+    return false if color == 'black' && y_position > y
+    false
```

Comment:
Since line 20 (`return true if dx.zero? && dy == 1`) will return from the method, if that passes it won't check the rest of the checks.  You can actually remove line 20 and replace this last `false` with the condition from 20, so the last line would just become:

```ruby
dx.zero? && dy == 1
```

That way it's the last check executed, and whatever its result is will be returned assuming everything else passes.  Or you could just reorder line 20 so that it's right before a final boolean:

```ruby
def valid_move?(x, y)
  ...
  return false unless dx.zero? && dy == 1
  true
end
```